### DISCUSSION TITLE
Possibility of modifying gcode in ZIP format

### DISCUSSION THREAD

**Discussion by cmsimon08** [sn4k3/UVtools#191] [Q&A] [answered]
This question may not be related to the software itself, but I'm just curious if this might work within the scope of this software.

I'm trying to modify the printing sequence (movement of z stage) to improve printing resolution. I've looked into possible ways it'd work and finally, found that one possible way is to directly modify gcode file generated when I save the slice file into ChituZIP format where basically the slicing file contains multiple PNG images and run.gcode.

![png_gcode](https://user-images.githubusercontent.com/69887978/116830627-7569a500-ab60-11eb-8444-35426335cf34.PNG)

So, I tried to re-write this gcode and re-zip the file on my PC. However, when I re-ZIP this file and import into UVtools, it'd not recognize this regenerated ZIP file. 

![import error](https://user-images.githubusercontent.com/69887978/116830935-7818ca00-ab61-11eb-83ba-fcb0df1559f7.PNG)

Which is no surprises since this happens in other slicers too (https://forum.chitubox.com/?thread-277.htm).

So, I was wondered if something like this (modifying gcode) is intrinsically impossible in any of slicing formats or something plausible with UVtools. Any knowledge or comment would be appreciated since I'm almost on a blank page how to do so.

Cheers,

**Comment by sn4k3** [comment, marked as answer]
UVtools don't allow modify gcode inside the GUI to prevent users from messing up, it also have it own gcode render and parser.
But you can modify the gcode after the file is generated, by open the zip, modify the gcode text file and update into zip. Nothing wrong doing that way.

About the error, if it saying Layer 1 not found, is because the '1.png' does not exists on zip, check that.

PS: You can use UVtools scripting to do automated stuff inside, but requires programming skills
